packages/terra-draw/src/modes/base.mode.ts

```diff
@@ -137,6 +138,10 @@ export abstract class TerraDrawBaseDrawMode<Styling extends CustomStyling> {
 		if (options?.pointerEvents !== undefined) {
 			this.pointerEvents = options.pointerEvents;
 		}
+
+		if (options?.modeName) {
+			this.mode = options.modeName;
+		}
```

Comment:
I was thinking about this and how do we handle the scenario where someone causes updateOptions on a mode after it's been initially constructed? The problem I'm foreseeing is if you change the name of the mode, but all the features in the store already have the original mode name, they will all break as there will be no mode associated with them any more.

I wonder if the modeName should be immutable after initialisation to avoid potential issues.